### DISCUSSION TITLE
get the curren LogLevel

### DISCUSSION THREAD

**Discussion by LuckyTurtleDev** [rwf2/Rocket#2275] [Questions] [answered]
Can I get the current [`LogLevel`](https://api.rocket.rs/v0.5-rc/rocket/config/enum.LogLevel.html)?

**Comment by SergioBenitez** [comment, marked as answer]
[`Rocket::config()`](https://api.rocket.rs/v0.5-rc/rocket/struct.Rocket.html#method.config)[`.log_level`](https://api.rocket.rs/v0.5-rc/rocket/config/struct.Config.html#structfield.log_level)

**Reply by LuckyTurtleDev** [reply]
Thanks

**Reply by LuckyTurtleDev** [reply]
What is the best way to access the config struct from everywhere?
Is it possible to put the struct in a static Lazy or something similar?

**Reply by SergioBenitez** [reply]
No, that would cause all sorts of issues if you ever had more than one Rocket application running. Anywhere Rocket is running you can access the config via the `&Config` request guard. Alternatively, you can access the config before Rocket launches and store it wherever you want.

**Reply by LuckyTurtleDev** [reply]
The best way I have found to get global access to the loglevel is using `rocket::Config::figment()`:

```rust
static LOG_LEVEL: Lazy<LogLevel> = Lazy::new(|| {
	rocket::Config::figment()
		.extract_inner("log_level")
		.expect("failed to get log level")
});
```
